controllers/callhomeconfig_controller.go

```diff
@@ -139,41 +143,56 @@ func (r *CallHomeConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			fmt.Sprintf("unexpected error getting service account: %s", err),
 			ctrl.Result{})
 	}
-	if len(sa.Secrets) < 1 {
-		return retryLater(r, log, &chc, "serviceaccount",
-			namespacedName.Name, "does not have a token")
-	}
-	tokenSecretName := sa.Secrets[0]
-	var token corev1.Secret
-	if err := r.Get(ctx, types.NamespacedName{
+	var secretList corev1.SecretList
+	var kubeconfigSecret corev1.Secret
+	secretFound := false
+	if err := r.List(ctx, &secretList, &client.ListOptions{
 		Namespace: req.Namespace,
-		Name:      tokenSecretName.Name,
-	}, &token); err != nil {
-		if apierrors.IsNotFound(err) {
-			return retryLater(r, log, &chc, "token secret",
-				tokenSecretName.Name, "does not exist yet")
+	}); err != nil {
+		return retryLater(r, log, &chc, "secrets", req.Namespace, "cannot list secrets")
+	}
+	if len(secretList.Items) == 0 {
+		return retryLater(r, log, &chc, "secrets", req.Namespace, "no secrets in the namespace")
+	}
+	for _, secret := range secretList.Items {
+		annotations := secret.GetAnnotations()
+		if annotations != nil {
+			val, ok := annotations["kubernetes.io/service-account.name"]
+			if !ok {
+				// the map is not nil, but the required key isn't present, so skip it
+				continue
+			}
+			if val == chc.Spec.ServiceAccountName {
+				kubeconfigSecret = secret
+				secretFound = true
```

Comment:
Should also insert a `break` to break out of the loop.

Reply:
will do